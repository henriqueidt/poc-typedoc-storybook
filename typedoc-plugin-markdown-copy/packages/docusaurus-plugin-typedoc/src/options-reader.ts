import { Options, OptionsReader } from 'typedoc';

export class MarkdownThemeOptionsReader implements OptionsReader {
  name = 'markdown-theme-reader';
  readonly order = 1000;
  readonly supportsPackages = false;
  read(container: Options) {
    container.setValue('theme', 'markdown');
  }
}
