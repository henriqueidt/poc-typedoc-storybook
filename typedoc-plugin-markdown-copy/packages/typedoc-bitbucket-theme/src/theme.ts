import { MarkdownTheme } from 'typedoc-plugin-markdown';
import { Renderer } from 'typedoc';

export class BitbucketTheme extends MarkdownTheme {
  constructor(renderer: Renderer) {
    super(renderer);
  }
  toAnchorRef(reflectionId: string) {
    return 'markdown-header-' + reflectionId;
  }
}
