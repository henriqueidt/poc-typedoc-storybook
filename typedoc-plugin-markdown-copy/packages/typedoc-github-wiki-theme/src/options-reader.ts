import { Options, OptionsReader } from 'typedoc';

export class GithubWikiThemeOptionsReader implements OptionsReader {
  name = 'github-wiki-theme-reader';
  readonly order = 900;
  readonly supportsPackages = false;
  read(container: Options) {
    if (container.getValue('theme') === 'default') {
      container.setValue('theme', 'github-wiki');
    }
  }
}
