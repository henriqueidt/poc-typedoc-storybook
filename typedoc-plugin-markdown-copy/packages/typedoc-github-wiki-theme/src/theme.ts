import * as fs from 'fs';
import { DeclarationReflection, Renderer, RendererEvent } from 'typedoc';
import { MarkdownTheme } from 'typedoc-plugin-markdown';

export class GithubWikiTheme extends MarkdownTheme {
  constructor(renderer: Renderer) {
    super(renderer);

    this.entryDocument = 'Home.md';
    this.hideBreadcrumbs = true;

    this.listenTo(this.owner, {
      [RendererEvent.END]: this.writeSidebar,
    });
  }

  getRelativeUrl(url: string) {
    return encodeURI('../wiki/' + url.replace('.md', ''));
  }

  toUrl(mapping: any, reflection: DeclarationReflection) {
    return `${reflection.getFullName().replace(/\//g, '.')}.md`;
  }

  writeSidebar(renderer: any) {
    const parseUrl = (url: string) => '../wiki/' + url.replace('.md', '');
    const navigation = this.getNavigation(renderer.project);
    const navJson: string[] = [`## ${renderer.project.name}\n`];
    const allowedSections = ['Home', 'Modules', 'Namespaces'];
    navigation.children
      ?.filter(
        (navItem) =>
          !navItem.isLabel || allowedSections.includes(navItem.title),
      )
      .forEach((navItem) => {
        if (navItem.isLabel) {
          navJson.push(`\n### ${navItem.title}\n`);
          navItem.children?.forEach((navItemChild) => {
            const longTitle = navItemChild.title.split('.');
            const shortTitle = longTitle[longTitle.length - 1];
            navJson.push(
              `- [${shortTitle}](${parseUrl(encodeURI(navItemChild.url))})`,
            );
          });
        } else {
          const title =
            navItem.url === this.entryDocument ? 'Home' : navItem.title;
          navJson.push(`- [${title}](${parseUrl(navItem.url)})`);
        }
      });

    fs.writeFileSync(
      renderer.outputDirectory + '/_Sidebar.md',
      navJson.join('\n') + '\n',
    );
  }

  get globalsFile() {
    return this.entryPoints.length > 1 ? 'Modules.md' : 'Exports.md';
  }
}
