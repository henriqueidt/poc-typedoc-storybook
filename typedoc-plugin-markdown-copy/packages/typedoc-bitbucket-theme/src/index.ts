import { Application } from 'typedoc';
import { BitbucketTheme } from './theme';

export function load(app: Application) {
  app.renderer.defineTheme('bitbucket', BitbucketTheme);
}
