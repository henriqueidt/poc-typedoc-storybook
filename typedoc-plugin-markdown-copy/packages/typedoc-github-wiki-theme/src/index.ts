import { Application } from 'typedoc';
import { GithubWikiTheme } from './theme';
import { GithubWikiThemeOptionsReader } from './options-reader';

export function load(app: Application) {
  app.renderer.defineTheme('github-wiki', GithubWikiTheme);
  app.options.addReader(new GithubWikiThemeOptionsReader());
}
