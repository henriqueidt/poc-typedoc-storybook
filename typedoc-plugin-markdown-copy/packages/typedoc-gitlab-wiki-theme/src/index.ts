import { Application } from 'typedoc';

import { GitlabWikiTheme } from './theme';

export function load(app: Application) {
  app.renderer.defineTheme('gitlab-wiki', GitlabWikiTheme);
}
