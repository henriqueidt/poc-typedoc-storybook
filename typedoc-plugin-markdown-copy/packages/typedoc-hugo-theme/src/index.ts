import { Application } from 'typedoc';
import { HugoTheme } from './theme';

export function load(app: Application) {
  app.renderer.defineTheme('hugo', HugoTheme);
}
