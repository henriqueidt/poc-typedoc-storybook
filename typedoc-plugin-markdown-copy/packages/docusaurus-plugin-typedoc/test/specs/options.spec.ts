import { getPluginOptions } from '../../dist/options';
import { PluginOptions } from '../../src/types';

describe(`Options:`, () => {
  test(`should return default options`, () => {
    expect(
      getPluginOptions({ entryPoints: ['entrypoint.ts'] }),
    ).toMatchSnapshot();
  });
  test(`should merge plugins`, () => {
    expect(
      getPluginOptions({
        plugin: ['plugin-xyz'],
      }).plugin,
    ).toMatchSnapshot();
  });

  test(`should merge custom sidebar options`, () => {
    expect(
      getPluginOptions({
        sidebar: {
          fullNames: true,
          indexLabel: 'Custom index',
          readmeLabel: 'Custom readme',
        },
      } as PluginOptions).sidebar,
    ).toMatchSnapshot();
  });
});
