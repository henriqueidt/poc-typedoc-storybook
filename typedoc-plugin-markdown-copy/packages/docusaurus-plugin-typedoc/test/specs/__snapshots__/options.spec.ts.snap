// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options: should merge custom sidebar options 1`] = `
{
  "autoConfiguration": true,
  "categoryLabel": "API",
  "collapsed": true,
  "fullNames": true,
  "indexLabel": "Custom index",
  "position": null,
  "readmeLabel": "Custom readme",
}
`;

exports[`Options: should merge plugins 1`] = `
[
  "typedoc-plugin-markdown",
  "plugin-xyz",
]
`;

exports[`Options: should return default options 1`] = `
{
  "cleanOutputDir": true,
  "docsRoot": "docs",
  "entryDocument": "index.md",
  "entryPoints": [
    "entrypoint.ts",
  ],
  "frontmatter": undefined,
  "hideBreadcrumbs": true,
  "hideInPageTOC": true,
  "hideMembersSymbol": false,
  "hidePageTitle": true,
  "id": "default",
  "includeExtension": true,
  "indexSlug": undefined,
  "out": "api",
  "plugin": [
    "typedoc-plugin-markdown",
  ],
  "sidebar": {
    "autoConfiguration": true,
    "categoryLabel": "API",
    "collapsed": true,
    "fullNames": false,
    "indexLabel": undefined,
    "position": null,
    "readmeLabel": "Readme",
  },
  "theme": "docusaurus",
  "watch": false,
}
`;
